app/src/main/java/openfoodfacts/github/scrachx/openfood/jobs/SavedProductUploadWork.java

```diff
@@ -0,0 +1,30 @@
+package openfoodfacts.github.scrachx.openfood.jobs;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.work.ListenableWorker;
+import androidx.work.WorkerParameters;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient;
+
+public class SavedProductUploadWork extends ListenableWorker {
```

Comment:
> For most cases, we recommend using Worker, which offers a simple synchronous API that is executed on a pre-specified background thread.

From the docs.

Is there any reason for using `ListenableWorker`?

Reply:
> > For most cases, we recommend using Worker, which offers a simple synchronous API that is executed on a pre-specified background thread.
> 
> From the docs.
> 
> Is there any reason for using `ListenableWorker`?

Yes, we used `ListenableWorker` as we need to execute asynchronous rest API inside the worker.